packages/lib/gpcircuits/test/tuple.spec.ts

```diff
@@ -0,0 +1,190 @@
+import { expect } from "chai";
+import { WitnessTester } from "circomkit";
+import "mocha";
+import { poseidon2, poseidon3, poseidon4 } from "poseidon-lite";
+import {
+  computeTupleIndices,
+  hashTuple,
+  maxTupleArity,
+  requiredNumTuples,
+  TupleModuleInputNamesType,
+  TupleModuleInputs,
+  TupleModuleOutputNamesType,
+  TupleModuleOutputs
+} from "../src";
+import { circomkit } from "./common";
+import { PODValue, podValueHash } from "@pcd/pod";
+import { BABY_JUB_NEGATIVE_ONE } from "@pcd/util";
+
+describe("Tuple helpers should work", function () {
+  it("should compute the right number of required tuples for different input tuple arities", () => {
+    [
+      [2, 2, 1],
+      [3, 3, 1],
+      [4, 4, 1],
+      [2, 5, 4],
+      [3, 5, 2],
+      [4, 5, 2]
+    ].forEach((triple) =>
+      expect(requiredNumTuples(triple[0], triple[1])).to.equal(triple[2])
+    );
+  });
+
+  it("should compute the right maximum tuple arity representable by different parameters", () => {
+    [
+      [1, 2, 2],
+      [1, 3, 3],
+      [1, 4, 4],
+      [4, 2, 5],
+      [2, 3, 5],
+      [2, 4, 7]
+    ].forEach((triple) =>
+      expect(maxTupleArity(triple[0], triple[1])).to.equal(triple[2])
+    );
+  });
+
+  it("should compute the right tuple indices for different input tuples and parameters", () => {
+    [
```

Comment:
I like this pattern for iterating through test-cases.
One suggestion particularly for cases as large as this one is that it might be clearest to use objects with named fields rather than bare arrays.  That way you could have the 3 named args, and an `expectedResult`.